lorien/InfiniteCanvas/Tools/SelectionTool.gd

```diff
@@ -36,7 +36,7 @@ func _ready():
 
 # ------------------------------------------------------------------------------------------------
 func tool_event(event: InputEvent) -> void:
-	if event.is_action_pressed("copy_strokes") || event.is_action_pressed("duplicate_strokes"):
+	if event.is_action_pressed("copy_strokes", true) || event.is_action_pressed("duplicate_strokes", true):
```

Comment:
I think you missed the second argument of the method. Only the third argument is the `exact_match` bool. 
This is the method signature: `bool is_action_pressed (String action, bool allow_echo=false, bool exact_match=false)`

See: https://docs.godotengine.org/en/stable/classes/class_inputevent.html#class-inputevent-method-is-action-pressed

Reply:
What a nice pitfall! `Input.is_action_pressed` and `InputEvent.is_action_pressed`, indeed have different signatures! Did not even realize this. Thanks for double checking :-)

I will replace all the uses by Input.is_action_pressed, I think. That way everything uses the same function to check if a keystroke is pressed. I think that makes most sense.

Reply:
Actually, scratch that. The current way is nicer...

Reply:
Oh boy... wanted to fix this quickly, but I think that I found a bug in Godot that complicates things... Check this here:

~~~~~~~~.gd
# Modified SelectionTool.gd
func tool_event(event: InputEvent) -> void:
	print("ST", 
		event.to_string(), 
		event.is_action_pressed("copy_strokes"), " ", 
		event.is_action_pressed("duplicate_strokes"), " ", 
		event.is_action_pressed("duplicate_strokes", false), " ", 
		event.is_action_pressed("duplicate_strokes", true), " ", 
		event.is_action_pressed("duplicate_strokes", false, true), " ", 
		event.is_action_pressed("duplicate_strokes", true, true)
	)
	print("ST2", 
		event.to_string(), 
		event.is_action_pressed("duplicate_strokes", false, true),
		event.is_action_pressed("duplicate_strokes", true, true)
	)
	print("ST3", 
		event.to_string(), 
		event.is_action_pressed("copy_strokes"), " ", 
		event.is_action_pressed("duplicate_strokes"), " ", 
		event.is_action_pressed("duplicate_strokes", false), " ", 
		event.is_action_pressed("duplicate_strokes", true), " ", 
		event.is_action_pressed("duplicate_strokes", false, true), " ", 
		event.is_action_pressed("duplicate_strokes", true, true)
	)
~~~~~~~~

Which produces the output when pressing the keystroke:

~~~~~~~~
ST[InputEventKey:7259]False True True True True True
ST2[InputEventKey:7259]FalseFalse
ST3[InputEventKey:7259]False True True True True True
ST[InputEventKey:7260]False False False False False False
ST2[InputEventKey:7260]FalseFalse
ST3[InputEventKey:7260]False False False False False False
~~~~~~~~

Seems like a bug to me, or am I missing something? `is_action_pressed` should probably return true for ST2 as well. (Sorry, a bit of a messy log, but I am a bit confused myself, still.

Reply:
I tried to isolate the bug yesterday. It, indeed, is a bug in Godot. I created a small test-project that shows that the bug occurs when using to_string() and seemignly also some other circumstances. I am not quite there yet. To finish this issue, I implemented a workaround for now, since we need to wait for another release anyway to "properly" do this :-/

See my last commits. I chanced everything to use the workaround version because it actually checks the events themselves rather than the "state" (which might occur more often than a key-press event itself).

Reply:
I am planning to report the bug to Godot, but correlating it with the amount of issues that are on _that_ repository will take some more time. I wanted to finish this work first.

Reply:
Alright, thanks! Well that's a really weird bug then, you would expect `to_string` to be const and not to change any internal values...
Yeah, it takes some time but _eventually_ someone fixes it^^

Reply:
Ok... I just took it on me and built the current Godot dev-branch from source to check if the bug is present on there as well, but as it turns out it is already fixed on there!

So we pretty much only need to wait for a release >3.4.4 and can replace the current workaround with `event.is_action_pressed("...", true, true)` which is equivalent to the current behavior.

Reply:
Hey there! Yeah, I was planning to dig at least to see why this is happening. As you said: `to_string()` modifying the state of Event did not seem correct, but also seemed like something that should be easy to detect and fix. I have some experience in working low-elvel on game engines already, but well, no need for doing the work if someone else has done it already...